src/app/user/actions/user.actions.ts

```diff
@@ -0,0 +1,28 @@
+import { createAction, props } from '@ngrx/store';
+import { User } from '../../../entities/user.model';
+
+const userActionNames = {
+  LOGIN: '[User] Login',
+  LOGIN_SUCCESS: '[User] Login succeeded',
+  LOGIN_FAIL: '[User] Login failed',
+  LOGOUT: '[User] Logout',
+  LOGOUT_SUCCESS: '[User] Logout succeeded',
+  LOGOUT_FAIL: '[User] Logout failed',
+  REGISTER: '[User] Register',
+  REGISTER_SUCCESS: '[User] Register succeeded',
+  REGISTER_FAIL: '[User] Register failed',
+};
+
+export const login = createAction(userActionNames.LOGIN, props<{ email: string; password: string; }>());
+export const loginSuccess = createAction(userActionNames.LOGIN_SUCCESS, props<{ user: User }>());
+export const loginFail = createAction(userActionNames.LOGIN_FAIL, props<{ message: string; }>());
+
+export const logout = createAction(userActionNames.LOGOUT);
+export const logoutSuccess = createAction(userActionNames.LOGOUT_SUCCESS);
+export const logoutFail = createAction(userActionNames.LOGOUT_FAIL, props<{ message: string; }>());
+
+export const register = createAction(userActionNames.REGISTER,
+  props<{ user: Omit<User, 'id'> & { password: string; } }>()
```

Comment:
what does it do? - & { password: string; } 
I got the part that omits the id key from the User.

Reply:
the User interface doesn't have the user password in it
while registering I want to get all the user details and the password.
This type requires the password as well, as part of the object